pkg/controller/operators/catalog/operator.go

```diff
@@ -1017,17 +1020,36 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 			return err
 		}
 		updatedSubs = o.setIPReference(updatedSubs, maxGeneration+1, installPlanReference)
-		for _, updatedSub := range updatedSubs {
-			for i, sub := range subs {
-				if sub.Name == updatedSub.Name && sub.Namespace == updatedSub.Namespace {
-					subs[i] = updatedSub
-				}
-			}
-		}
 	} else {
 		logger.Debugf("no subscriptions were updated")
 	}
 
+	// Remove resolutionfailed condition from subscriptions
+	subs = o.removeSubsCond(subs, v1alpha1.SubscriptionResolutionFailed)
+	newSub := true
+	for _, updatedSub := range updatedSubs {
```

Comment:
suggestion: maybe it's worth moving this to its own function? Then rather than mutating the subs slice, just return a new slice?

Reply:
+1 to this suggestion, although it feels like this method as a whole may need to be refactored at some point as there's a good chunk of complexity now. maybe we could tackle the refactoring as a follow-up?